rdmo/projects/utils.py

```diff
@@ -269,3 +271,64 @@ def compute_set_prefix_from_set_value(set_value, value):
         str(set_value.set_index) if (index == set_prefix_length) else value
         for index, value in enumerate(value.set_prefix.split('|'))
     ])
+
+
+def get_contact_message(request, project):
+    project_url = project.get_absolute_url()
+    project_interview_url = reverse('project_interview', args=[project.id])
+
+    context = {
+        'user': request.user,
+        'site': Site.objects.get_current(),
+        'project': project,
+        'project_url': request.build_absolute_uri(project_url)
+    }
+
+    page_id = request.GET.get('page')
+    if page_id:
+        page = project.catalog.get_page(page_id)
+        if page:
+            context.update({
+                'page': page,
+                'page_url': request.build_absolute_uri(f'{project_interview_url}{page_id}/')
+            })
+
+    questionset_id = request.GET.get('questionset')
+    if questionset_id:
+        context['questionset'] = project.catalog.get_questionset(questionset_id)
+
+    question_id = request.GET.get('question')
+    if question_id:
+        context['question'] = project.catalog.get_question(question_id)
+
+    value_ids = request.GET.getlist('values')
+    if value_ids:
+        values = project.values.filter(snapshot=None).filter(id__in=value_ids)
```

Comment:
Could somehow the order of checkbox question answers (options) be preserved? 
```py
.order_by('option__option_optionsets__order')
```

Reply:
I think it could be fixed here: https://github.com/rdmorganiser/rdmo/blob/interview-contact-modal/rdmo/projects/assets/js/interview/components/main/widget/CheckboxWidget.js#L17 by looping over the existing values and updating them with a new `collection_index` but I would fix this independent of this PR.

Reply:
alright!